observability/akka-http-metrics/src/test/suite/scala/com/daml/metrics/akkahttp/WebSocketMetricsSpec.scala

```diff
@@ -0,0 +1,342 @@
+
+  private val getMessagesReceivedBytesTotalValue: TestMetrics => Long =
+    (metrics: TestMetrics) => metrics.messagesReceivedBytesTotalValue
+
+  private val getMessagesSentBytesTotalValue: TestMetrics => Long = (metrics: TestMetrics) =>
+    metrics.messagesSentBytesTotalValue
+
+  private def checkCountThroughFlow(
+      messages: List[Message],
+      metricExtractor: (TestMetrics) => Long,
+      expected: Long,
+  ): Future[Assertion] = {
+    withDuplicatingFlowAndMetrics { (flow, metrics) =>
+      val done = Source(messages).via(flow).runWith(drainStreamedMessages)
+      done.map { _ =>
+        metricExtractor(metrics) should be(expected)
+      }
+    }
+  }
+
+  private def checkCountThroughFlow(
+      message: Message,
+      metricExtractor: TestMetrics => Long,
+      expected: Long,
+  ): Future[Assertion] = {
+    checkCountThroughFlow(List(message), metricExtractor, expected)
+  }
+
+  "websocket_message_received_total" should {
+    "count passing strict text messages" in {
+      checkCountThroughFlow(strictTextMessage, getMessagesReceivedTotalValue, 1)
+    }
+
+    "count passing streamed text messages" in {
+      checkCountThroughFlow(streamedTextMessage, getMessagesReceivedTotalValue, 1)
+    }
+
+    "count passing strict binary messages" in {
+      checkCountThroughFlow(strictBinaryMessage, getMessagesReceivedTotalValue, 1)
+    }
+
+    "count passing streamed binary messages" in {
+      checkCountThroughFlow(streamedBinaryMessage, getMessagesReceivedTotalValue, 1)
+    }
+
+    "count a set of passing messages" in {
+      checkCountThroughFlow(
+        List[Message](
+          strictTextMessage,
+          streamedTextMessage,
+          strictBinaryMessage,
+          streamedBinaryMessage,
+        ),
+        getMessagesReceivedTotalValue,
+        4,
+      )
+    }
+  }
+
+  "websocket_message_sent_total" should {
+    "count passing strict text messages" in {
+      checkCountThroughFlow(strictTextMessage, getMessagesSentTotalValue, 2)
+    }
+
+    "count passing streamed text messages" in {
+      checkCountThroughFlow(streamedTextMessage, getMessagesSentTotalValue, 2)
+    }
+
+    "count passing strict binary messages" in {
+      checkCountThroughFlow(strictBinaryMessage, getMessagesSentTotalValue, 2)
+    }
+
+    "count passing streamed binary messages" in {
+      checkCountThroughFlow(streamedBinaryMessage, getMessagesSentTotalValue, 2)
+    }
+
+    "count a set of passing messages" in {
+      checkCountThroughFlow(
+        List[Message](
+          strictTextMessage,
+          streamedTextMessage,
+          strictBinaryMessage,
+          streamedBinaryMessage,
+        ),
+        getMessagesSentTotalValue,
+        8,
+      )
+    }
+  }
+
+  "websocket_message_received_bytes_total" should {
+    "count size of passing strict text messages" in {
+      checkCountThroughFlow(
+        strictTextMessage,
+        getMessagesReceivedBytesTotalValue,
+        6L,
```

Comment:
Can we make these magic constants explicit, please? I.e., I'd suggest either creating separate constants for the sizes associated to the different input message types, or, even better, to couple these two pieces of information via a case class so that e.g. you can call `strictTextMessage.value` and `strictTextMessage.size`.